src/libsystemd-network/sd-dhcp6-client.c

```diff
@@ -1160,6 +1170,10 @@ static int client_process_advertise_or_rapid_commit_reply(
         if (message->type == DHCP6_MESSAGE_REPLY) {
                 bool rapid_commit;
 
+                if (!client->rapid_commit)
+                        return log_dhcp6_client_errno(client, SYNTHETIC_ERRNO(EINVAL),
+                                                      "Received unexpected reply message, ignoring.");
```

Comment:
maybe mention why unexpected in the log msg

Reply:
Updated.